src/mantle/support/helpers/helpers-general.php

```diff
@@ -428,3 +428,50 @@ function hook_callable( string $hook, callable $callable, int $priority = 10 ):
 		$callable();
 	}
 }
+
+/**
+ * Validates a file name and path against an allowed set of rules.
+ *
+ * A return value of `1` means the file path contains directory traversal.
+ *
+ * A return value of `3` means the file is not in the allowed files list.
+ *
+ * @see validate_file() in WordPress core.
+ *
+ * @param string   $file          File path.
+ * @param string[] $allowed_files Optional. Array of allowed files. Default empty array.
+ * @return int 0 means nothing is wrong, greater than 0 means something was wrong.
+ */
+function validate_file( $file, $allowed_files = [] ) {
+	// Proxy back to the core function if it exists allowing Window drive paths.
+	if ( function_exists( 'validate_file' ) ) {
+		return in_array( \validate_file( $file, $allowed_files ), [ 0, 2 ], true ) ? 0 : 1;
```

Comment:
`validate_file` can also return `3`, which means a mismatch against the allowed files list:
```suggestion
		$retval = \validate_file( $file, $allowed_files );
		return in_array( $retval, [ 0, 2 ], true ) ? 0 : $retval;
```